pkg/hive/hive.go

```diff
@@ -284,12 +284,16 @@ func (s *Service) checkAndAddPeers(ctx context.Context, peers pb.Peers) {
 			ctx, cancel := context.WithTimeout(ctx, pingTimeout)
 			defer cancel()
 
+			start := time.Now()
+
 			// check if the underlay is usable by doing a raw ping using libp2p
 			if _, err = s.streamer.Ping(ctx, multiUnderlay); err != nil {
+				s.metrics.PingFailureTime.Observe(float64(time.Since(start).Milliseconds()))
```

Comment:
it is not needed to convert to milliseconds. grafana supports nanoseconds like the default golang time representation